include/mgba/core/interface.h

```diff
@@ -38,7 +38,7 @@ typedef uint32_t color_t;
 #ifndef PYCPARSE
 static inline color_t mColorFrom555(uint16_t value) {
 #ifdef COLOR_16_BIT
-#ifdef COLOR_5_6_5
```

Comment:
Um, why? This seems like a bad idea.

Reply:
RetroArch version of PS2 is not implementing color correction in the RetroArch itself.
The colour correction is an expensive operation and for consoles less powerful there is a huge difference.

So in this case I just returning the value that PS2 expect to receive

Reply:
But that will result in the colors being flat out wrong. The lowest bit of green will become the top bit of blue, making slightly green things super blue.

Reply:
Unless you mean the PS2 uses 555, in which case you shouldn't be defining 565.

Reply:
@endrift This is the only way to get it to perform properly on PS2 right now. The colors are all messed up with 565, and performance is noticeably lower.

I believe the PS2 texture format being used is natively 555, so if he were to use 565, it would still be implicitly copied/converted to 555 anyway. There is not the luxury to do such expensive operations on PS2.

Reply:
Hi again.
So step by step,
Yes PS2 supports BGR555.
About as you said don't define 565, you are right. It was my first approach, but then I noticed that colours didn't worked properly.
So either, the core had something wrong for 555 or the core skip some places to do the proper colour correction.
I was investigating but I didn't find anything wrong.

I can attach tomorrow specific images

So then, testing I saw that just making this change the colours were right for gbc and gba.
For me it sounds weird as well but I couldn't find a better way to do it.

I hope that this could bring more light to the conversation

Reply:
Did you get those images? I set up a PS2 toolchain and can compile mGBA with it now, but I don't seem to be able to load ROMs in PCSX2 so I'll need to hook up my PS2 to actually figure out what to do with this.

Reply:
I will attach them later

Reply:
Hello @endrift,
So here you are:

The first image is the current status, with the workaround that I did:

<img width="662" alt="screen shot 2019-03-04 at 23 03 19" src="https://user-images.githubusercontent.com/10010409/53766279-b4453700-3ed2-11e9-9019-eed954a91042.png">

However, the next image is after remove the `-DCOLOR_5_6_5` flag.
<img width="662" alt="screen shot 2019-03-04 at 23 07 52" src="https://user-images.githubusercontent.com/10010409/53766375-f4a4b500-3ed2-11e9-9f7b-7c8dd0fd0a94.png">

If you notice there are blues that are changed to black but is quite fine because I saw other blues that are exactly the same.
I think that this probably will require a deeper investigation, and maybe we should open a specific issue to track this info.

Thanks

Reply:
If you have the tipical rom with all the different colors and so on, it is going to be easier to identify what are colours that are right, and what are the colours that are wrong.
If you have this rom, please send to me

Reply:
Try backing out that change, removing the ifdef and making it use the pixel format 565 flag in libretro.c? I'm not sure how you are able to run a ROM inside PCSX2; it always hangs when I try to select a memory card. Maybe I need to format it somehow?

Reply:
Hello,
PCSX2 is somehow not the best tool to give a try.
So here you are the image, just keeping 565 and removing my workaround.

<img width="662" alt="screen shot 2019-03-05 at 00 42 38" src="https://user-images.githubusercontent.com/10010409/53770572-e0b38000-3edf-11e9-9285-3e9c8f4940f4.png">

Obviously you can notice how, red are blues and blues are red, is the difference between RGB565 and BGR555, (actually it is ABGR1555)

Finally here you are a mermory card "prepared", just unzip the file and insert in the port 0 of the MCs.

[memory_mgba_ready.zip](https://github.com/libretro/mgba/files/2928638/memory_mgba_ready.zip)

Thanks

Reply:
If you back out the mColor change and do this instead it should work equivalently and non-invasively:

```patch
 void retro_init(void) {
        enum retro_pixel_format fmt;
 #ifdef COLOR_16_BIT
-#ifdef COLOR_5_6_5
+#if defined(COLOR_5_6_5) || defined(PS2)
        fmt = RETRO_PIXEL_FORMAT_RGB565;
 #else
```

Reply:
I don't think so, these values are used in RetroArch and I'm ignoring them

Reply:
I forgot to mention, that's with 5_6_5 not being defined anymore.

Reply:
As I was thinking, it behave exactly the same.
I'm pretty sure that mgba is not supporting properly BGR555, you have some issue somewhere.

Reply:
That doesn't sound right. If you disable 5_6_5 it shouldn't ever swap the color channels, which is what your patch changed.

Reply:
The line that you wanted to change is just affecting the iplementation of RetroArch. 
I’m not doing color correction in RetroArch (for performance reasons) and I’m just checking if it 32bits or 16bits.
So, I wanted to say that it behave exactly the same, but anyway I wll try it later.